src/neps/api.py

```diff
@@ -106,7 +108,7 @@ def run(
     ignore_errors: bool = False,
     loss_value_on_error: None | float = None,
     cost_value_on_error: None | float = None,
-    pre_load_hooks: List=[],
+    pre_load_hooks: List = [],
```

Comment:
Some information about typing, don't need to do anything here but thought it mgiht be good to share:

1. If you have `from __future__ import annotations` (which there is), you can use the lowercase `list`
2. [Do not use mutable arguments as a default](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments).
3. We can type it better by following down where they're used, from what I gather it accepts `Callable[[metahyper.Sampler], metahyper.Sampler]`, where it can modify the sampler. Consider the typing can happen at a `neps` level and we only care about `class BaseOptimizer(metahyper.Sampler): ...`, then the full signature might look like:

```python
def run(
    ...,
    pre_load_hooks: list[Callable[[BaseOptimizer], BaseOptimizer]] | None = None
):
    if pre_load_hooks is None:
        pre_load_hooks = []
```

4. We can actually improve it even further using `typing.Iterable` to mean anything that we can iterate

```python
from typing import Iterable

def run(
    ...,
    pre_load_hooks: Iterable[Callable[[BaseOptimizer], BaseOptimizer]] | None = None
):
    if pre_load_hooks is None:
        pre_load_hooks = []
```

This means someone could pass a `tuple`, `set`, `generator` or really anything where we can go `for x in my_iterable: ...`.

5. Last bit is definitely over the top and low priority, but just to share some new typing tips. The function is actually a bit more specific in that it should return the **same** type of `BaseOptimizer` that went into it, not just any base optimizer. We can use `TypeVar`'s for this, which are essentially C+++ Templates or Java Generics.

```python
from typing import TypeVar

# This is some type that must be subclass of BaseOptimizer
OptimizerT = TypeVar("OptimizerT", bound=BaseOptimizer)

def run(
    ...,
    pre_load_hooks: Iterable[Callable[[OptimizerT], OptimizerT]] | None = None
):
    if pre_load_hooks is None:
        pre_load_hooks = []
```

@Neeratyoy thought you might also like to know about these things if you are reviewing